internal/pkg/cli/task_run.go

```diff
@@ -69,13 +79,19 @@ func newTaskRunOpts(vars runTaskVars) (*runTaskOpts, error) {
 		return nil, fmt.Errorf("new config store: %w", err)
 	}
 
-	return &runTaskOpts{
+	sess, err := session.NewProvider().Default()
```

Comment:
If a user provides an environment after `Ask` or with flags, we should instead use the environment manager role to create the task stack:
https://github.com/aws/copilot-cli/blob/2fb935f2fda9251dceeee0f72b2d77b602ba55d3/internal/pkg/cli/svc_deploy.go#L255
This is a role that Copilot creates when a user runs `copilot env init` for developers to assume and create/monitor their services or tasks.

-----------

The way we've been initializing clients that require data after the `Ask` method is with function fields, for example:
https://github.com/aws/copilot-cli/blob/2fb935f2fda9251dceeee0f72b2d77b602ba55d3/internal/pkg/cli/svc_logs.go#L74-L84

The reason is that in our tests we can override the `initCwLogsSvc` field to return a mock instead. This way we don't need to define global variables in the code.

Reply:
For the second one (initializing clients): that's exactly what I did!  it's coming in the next PR.

Reply:
I actually have coded everything for `task_run.go` but the PR got too large, so I split it into three smaller ones. This is the first of the smaller PR.

Reply:
I will use `envSession` to configure runtime clients in the next PR (since this PR doesn't involve any runtime clients yet)